src/app/[language]/page.tsx

```diff
@@ -0,0 +1,55 @@
+import Link from 'next/link'
+import { useTranslation } from '../i18n'
+
+import { languages, fallbackLanguage } from '../i18n/settings'
+
+interface Props {
+  params: {
+    language: string
+  }
+}
+
+
+export default async function Page({ params }: Props) {
+  let language = params?.language
+  if (languages.indexOf(language) < 0) language = fallbackLanguage
+  const { t } = await useTranslation(language, 'page')
```

Comment:
Let's refactor this hook so it fetches the language inside the hook because it can be quite cumbersome to always fetch the language on every page.

You can look at `useVisaFormProgress` for reference on how this can be done. Although you probably won't need zod for a single param.